common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/contests/redux/requests/ContestsRequests.kt

```diff
@@ -1,22 +1,43 @@
 package io.xorum.codeforceswatcher.features.contests.redux.requests
 
+import com.soywiz.klock.DateTime
 import io.xorum.codeforceswatcher.features.contests.models.Contest
+import io.xorum.codeforceswatcher.features.contests.models.Platform
 import io.xorum.codeforceswatcher.network.CodeforcesApiClient
+import io.xorum.codeforceswatcher.network.KontestsApiClient
 import io.xorum.codeforceswatcher.redux.Message
 import io.xorum.codeforceswatcher.redux.Request
 import io.xorum.codeforceswatcher.redux.ToastAction
 import io.xorum.codeforceswatcher.redux.store
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
 import tw.geothings.rekotlin.Action
 
 class ContestsRequests {
 
     class FetchContests(val isInitiatedByUser: Boolean) : Request() {
 
         override suspend fun execute() {
-            val response = CodeforcesApiClient.getContests()
-            response?.result?.let { contests ->
-                store.dispatch(Success(contests))
-            } ?: dispatchFailure()
+            val responseCodeforces = CoroutineScope(Dispatchers.Main).async { CodeforcesApiClient.getCodeforcesContests() }
+            val responseKontests = CoroutineScope(Dispatchers.Main).async { KontestsApiClient.getAllContests() }
+
+            val codeforcesContests = responseCodeforces.await()?.result
+            val allContests = responseKontests.await()
+
+            if (codeforcesContests == null && allContests == null) {
+                dispatchFailure()
+                return
+            }
+            val contests: MutableList<Contest> = mutableListOf()
```

Comment:
Think about the format, where you can write `contests = fetchAllContests() + fetchCodeforcesContests()`. It will be better in terms of readability.